tests/core/unittests/mongodb_test.py

```diff
@@ -60,12 +60,14 @@ def test_bad_connection(self):
             MongoDB(host='asdfada', port=123, dbname='metaopt',
                     username='uasdfaf', password='paasdfss')
         assert "Connection" in str(exc_info.value)
+        MongoDB.instance = None  # Set singular instance to None for independent tests
```

Comment:
Isn't there something similar to TestCast.tearDown() in pytest?

Reply:
There is, I fix this in the next PR. About experiment and trials.

Reply:
Why not fixing it here?

Reply:
I am grouping "session" setup/teardowns in a default file `conftest.py` by pytest. I did that in another branch.